Send Redirect to Relative URL

pmd1nh opened the pull request:
https://github.com/OpenLiberty/open-liberty/issues/10000

pmd1nh commented on dev/com.ibm.ws.webcontainer/resources/OSGI-INF/l10n/metatype.properties:
```diff
@@ -174,3 +174,6 @@ getRealPathReturnsQualifiedPath.desc=Return non-null paths from the ServletConte
 stopAppStartUponListenerException.desc=Some web applications depend on context listeners for setup before the web application starts. Set this property to true to stop the application from starting up when an unhandled exception is thrown from the context listeners.
+
+redirectToRelativeUrl.name=Send response redirect to a relative URL location
+redirectToRelativeUrl.desc=Send response redirect to a relative URL location without processing it. Set this property to true to send response redirect without checking and converting to an absolute URL location. 
```
sounds good

dmuelle replied:
based on above comments, should be `redirect response`. I'm also wondering if "checking" is needed here, or if it might be part of "converting". If not:

`Set this property to true to send a redirect response without converting the URL to an absolute location.`

pmd1nh replied:
sounds good!

pmd1nh commented on dev/com.ibm.ws.webcontainer/resources/OSGI-INF/l10n/metatype.properties:
```diff
@@ -174,3 +174,6 @@ getRealPathReturnsQualifiedPath.desc=Return non-null paths from the ServletConte
 stopAppStartUponListenerException.name=Stop application for unhandled listener exception
 stopAppStartUponListenerException.desc=Some web applications depend on context listeners for setup before the web application starts. Set this property to true to stop the application from starting up when an unhandled exception is thrown from the context listeners.
+
+redirectToRelativeUrl.name=Send response redirect to a relative URL location
```
I think "redirect response" is more accurate.

dmuelle replied:
ok in that case
**change**
Send response redirect to a relative URL location
**to**
Send a redirect response to a relative URL location

pmd1nh commented:
Updated with David's comments

pmd1nh commented:
#build #spawn.fullfat.buckets=com.ibm.ws.webcontainer_fat,com.ibm.ws.webcontainer.servlet.3.1_fat,com.ibm.ws.webcontainer.servlet.4.0_fat

pnicolucci commented:
@pmd1nh  what do we have remaining here?

pmd1nh commented:
#build #spawn.fullfat.buckets=com.ibm.ws.webcontainer_fat,com.ibm.ws.webcontainer.servlet.3.1_fat,com.ibm.ws.webcontainer.servlet.4.0_fat

pmd1nh commented:
PB results: 1 error is unrelated.

Tests: |   | 67612 tests, 0 failures, 1 errors
-- | -- | --

pnicolucci commented:
#build

pmd1nh commented:
rebased and 

#build

pmd1nh commented:
rebased, then rebuild both OL and CL (also rebased), then run local feature checker.  No Issue

========
   [delete] Deleting /ASLibertyGit/Mar30_18/WS-CD-Open/dev/build.image/wlp/lib/features/protected.20006.9.mf
     [echo] Feature check passed. No Issues Found.

BUILD SUCCESSFUL
Total time: 30 minutes 21 seconds

pmd1nh commented:
#build        #spawn.fullfat.buckets=com.ibm.ws.webcontainer_fat,com.ibm.ws.webcontainer.servlet.3.1_fat,com.ibm.ws.webcontainer.servlet.4.0_fat,com.ibm.ws.webcontainer_fat_lWAS,com.ibm.ws.webcontainer_fat_servlet31_webcontainerlWAS,com.ibm.ws.webcontainer.servlet.4.0_fat_toleration.3.0,com.ibm.ws.webcontainer.servlet.4.0_fat_toleration.3.0.lWAS

pmd1nh commented:
merged with latest integration change.  resubmit PB

#build #spawn.fullfat.buckets=com.ibm.ws.webcontainer_fat,com.ibm.ws.webcontainer.servlet.3.1_fat,com.ibm.ws.webcontainer.servlet.4.0_fat,com.ibm.ws.webcontainer_fat_lWAS,com.ibm.ws.webcontainer_fat_servlet31_webcontainerlWAS,com.ibm.ws.webcontainer.servlet.4.0_fat_toleration.3.0,com.ibm.ws.webcontainer.servlet.4.0_fat_toleration.3.0.lWAS

pnicolucci commented:
@pmd1nh we should have feature checker failures and require updates similar to what we had for https://github.com/OpenLiberty/open-liberty/pull/12085 and previous updates to WCCustomProperties that is an SPI. It does not look like feature checker failed in your PBs so we need to find out why. Are the feature checker baselines not updated every release or something of that nature? I think we need to talk with @fritze2

pmd1nh commented:
Feature checker does not fail in my local FC either. 
At this stage, the PR will miss the 20.0.0.10 translation for the new metatype property.

I'm escalating the problem to @fritze2 in separate email.

pmd1nh commented:
rebased on the latest OL and CL.  Local feature check had error before I updated bnd version.  It is now good.  run PB.

#build #spawn.fullfat.buckets=com.ibm.ws.webcontainer_fat,com.ibm.ws.webcontainer.servlet.3.1_fat,com.ibm.ws.webcontainer.servlet.4.0_fat,com.ibm.ws.webcontainer_fat_lWAS,com.ibm.ws.webcontainer_fat_servlet31_webcontainerlWAS,com.ibm.ws.webcontainer.servlet.4.0_fat_toleration.3.0,com.ibm.ws.webcontainer.servlet.4.0_fat_toleration.3.0.lWAS

pmd1nh commented:
#build #spawn.fullfat.buckets=com.ibm.ws.webcontainer_fat,com.ibm.ws.webcontainer.servlet.3.1_fat,com.ibm.ws.webcontainer.servlet.4.0_fat,com.ibm.ws.webcontainer_fat_lWAS,com.ibm.ws.webcontainer_fat_servlet31_webcontainerlWAS,com.ibm.ws.webcontainer.servlet.4.0_fat_toleration.3.0,com.ibm.ws.webcontainer.servlet.4.0_fat_toleration.3.0.lWAS

pnicolucci commented on dev/com.ibm.ws.webcontainer/src/com/ibm/wsspi/webcontainer/WCCustomProperties.java:
```diff
@@ -316,6 +316,9 @@
     public static boolean DECODE_URL_PLUS_SIGN;
 
+    //20.0.0.12
+    public static boolean REDIRECT_TO_RELATIVE_URL;
```
Phu I think we need to update this comment to reflect whatever release this is going to make it into. Please check it out, update the comment, run a build and lets get this merged this week if we can.

pnicolucci commented on dev/com.ibm.ws.webcontainer/src/com/ibm/wsspi/webcontainer/WCCustomProperties.java:
```diff
@@ -792,6 +795,9 @@ public static void setCustomPropertyVariables() {
         GET_REAL_PATH_RETURNS_QUALIFIED_PATH = Boolean.valueOf(WebContainer.getWebContainerProperties().getProperty("com.ibm.ws.webcontainer.getrealpathreturnsqualifiedpath", "true")).booleanValue();
         
+        //20.0.0.12
+        REDIRECT_TO_RELATIVE_URL = Boolean.valueOf(WebContainer.getWebContainerProperties().getProperty("com.ibm.ws.webcontainer.redirecttorelativeurl", "false")).booleanValue();
```
Same goes for this comment

pmd1nh commented:
Corrected the release version for this fix.

#build #spawn.fullfat.buckets=com.ibm.ws.webcontainer_fat,com.ibm.ws.webcontainer.servlet.3.1_fat,com.ibm.ws.webcontainer.servlet.4.0_fat,com.ibm.ws.webcontainer_fat_lWAS,com.ibm.ws.webcontainer_fat_servlet31_webcontainerlWAS,com.ibm.ws.webcontainer.servlet.4.0_fat_toleration.3.0,com.ibm.ws.webcontainer.servlet.4.0_fat_toleration.3.0.lWAS

pmd1nh commented:
#build #spawn.fullfat.buckets=com.ibm.ws.webcontainer_fat,com.ibm.ws.webcontainer.servlet.3.1_fat,com.ibm.ws.webcontainer.servlet.4.0_fat,com.ibm.ws.webcontainer_fat_lWAS,com.ibm.ws.webcontainer_fat_servlet31_webcontainerlWAS,com.ibm.ws.webcontainer.servlet.4.0_fat_toleration.3.0,com.ibm.ws.webcontainer.servlet.4.0_fat_toleration.3.0.lWAS

pmd1nh commented:
#run-libby-bot
